File: Lecture 12/F12 CSH/F12.cs

using static System.Console;

internal class F12
{
class Vehicle { }
class Car : Vehicle { }
class Bike : Vehicle { }
static void Main(string[] _)
{
    Vehicle[] a= new Vehicle[100];
    int iv=0, ch, nCars=0;
    do
    {
        Write("1=Car,2=Bike,0=Quit :"); 
        ch = Convert.ToInt32(ReadLine());
        if (ch == 1) a[iv++] = new Car();
        if (ch == 2) a[iv++] = new Bike();
    } while (ch != 0);

    
    for (int i=0; i < iv; i++)
    {
        Vehicle v = a[i];
        WriteLine(v.ToString());
        if (v is Car)
            nCars++;
    }
    WriteLine($"Cars: {nCars}");
}

}

File: Lecture 12/F12 CPP/F12.cpp
#include <iostream> 
#include <typeinfo>
using namespace std;
class Vehicle { virtual void x() {}};
class Car : public Vehicle { };
class Bike : public Vehicle { };

int main() {
	Vehicle* a[100];  //  [1020, 1080, 1100]

	int iv = 0, ch = 0, nCars = 0;
	do {
		wcout <<  L"1=Car, 2=Bike, 0=Quit: ";
		wcin  >> ch;
		if (ch == 1) a[iv++] = new Car();
		if (ch == 2) a[iv++] = new Bike();
	} while (ch);

	for (int i = 0; i < iv; i++) {
		Vehicle* pv = a[i];
		wcout << typeid(*pv).name() << endl;
		if (dynamic_cast<Car*>(pv))
			nCars++;
	}
	wcout << "Cars:" << nCars;
	for (int i = 0; i < iv; i++)
		delete a[i];
}
